Fix NoUncountedMemberChecker warnings in WebScreenOrientationManagerProxy, WebModelPlayerProvider, and WebSpeechSynthesisClient.

rniwa opened the pull request:
#### 39d2925d59524e6104e93f5b1b08eb6b7e3dc781
<pre>
Fix NoUncountedMemberChecker warnings in WebScreenOrientationManagerProxy, WebModelPlayerProvider, and WebSpeechSynthesisClient.
<a href="https://bugs.webkit.org/show_bug.cgi?id=279286">https://bugs.webkit.org/show_bug.cgi?id=279286</a>

Reviewed by Chris Dumez.

Use more smart pointers for member variables in these classes.

* Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
(WebKit::WebScreenOrientationManagerProxy::WebScreenOrientationManagerProxy):
(WebKit::WebScreenOrientationManagerProxy::~WebScreenOrientationManagerProxy):
(WebKit::WebScreenOrientationManagerProxy::setCurrentOrientation):
(WebKit::WebScreenOrientationManagerProxy::lock):
(WebKit::WebScreenOrientationManagerProxy::unlock):
* Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.h:
* Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp:
(WebKit::WebModelPlayerProvider::createModelPlayer):
* Source/WebKit/WebProcess/Model/WebModelPlayerProvider.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.cpp:
(WebKit::WebSpeechSynthesisClient::WebSpeechSynthesisClient):
(WebKit::WebSpeechSynthesisClient::protectedPage const):
(WebKit::WebSpeechSynthesisClient::voiceList):
(WebKit::WebSpeechSynthesisClient::corePageObserver const):
(WebKit::WebSpeechSynthesisClient::resetState):
(WebKit::WebSpeechSynthesisClient::speak):
(WebKit::WebSpeechSynthesisClient::cancel):
(WebKit::WebSpeechSynthesisClient::pause):
(WebKit::WebSpeechSynthesisClient::resume):
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.h:
(WebKit::WebSpeechSynthesisClient::WebSpeechSynthesisClient): Deleted.

Canonical link: <a href="https://commits.webkit.org/283330@main">https://commits.webkit.org/283330@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/074c742b2e6a3f5d2ff257a60814592a0137aa17

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65984 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45357 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18603 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70014 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16593 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53156 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16875 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52960 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11542 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69051 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41856 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57126 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33595 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38527 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15469 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60403 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14850 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71716 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9939 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14263 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60276 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9971 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57188 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60566 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8210 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1850 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9984 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41165 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42241 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43424 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41985 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
```diff
@@ -45,14 +45,14 @@ WebScreenOrientationManagerProxy::WebScreenOrientationManagerProxy(WebPageProxy&
     , m_currentOrientation(orientation)
 {
-    m_page.legacyMainFrameProcess().addMessageReceiver(Messages::WebScreenOrientationManagerProxy::messageReceiverName(), m_page.webPageIDInMainFrameProcess(), *this);
+    m_page->legacyMainFrameProcess().addMessageReceiver(Messages::WebScreenOrientationManagerProxy::messageReceiverName(), m_page->webPageIDInMainFrameProcess(), *this);
```
`protectedLegacyMainFrameProcess()->` no?

rniwa replied:
oh sure, fixed.

cdumez commented on Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
```diff
@@ -45,14 +45,14 @@ WebScreenOrientationManagerProxy::WebScreenOrientationManagerProxy(WebPageProxy&
     unlockIfNecessary();
 
-    m_page.legacyMainFrameProcess().removeMessageReceiver(Messages::WebScreenOrientationManagerProxy::messageReceiverName(), m_page.webPageIDInMainFrameProcess());
+    m_page->legacyMainFrameProcess().removeMessageReceiver(Messages::WebScreenOrientationManagerProxy::messageReceiverName(), m_page->webPageIDInMainFrameProcess());
```
ditto.

rniwa replied:
fixed.

cdumez commented on Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
```diff
@@ -69,7 +69,7 @@ void WebScreenOrientationManagerProxy::setCurrentOrientation(WebCore::ScreenOrie
         return;
 
-    m_page.legacyMainFrameProcess().send(Messages::WebScreenOrientationManager::OrientationDidChange(orientation), m_page.webPageIDInMainFrameProcess());
+    m_page->legacyMainFrameProcess().send(Messages::WebScreenOrientationManager::OrientationDidChange(orientation), m_page->webPageIDInMainFrameProcess());
```
ditto.

rniwa replied:
fixed

cdumez commented on Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
```diff
@@ -122,15 +122,15 @@ void WebScreenOrientationManagerProxy::lock(WebCore::ScreenOrientationLockType l
-        if (m_page.fullScreenManager() && m_page.fullScreenManager()->isFullScreen()) {
-            if (!m_page.fullScreenManager()->lockFullscreenOrientation(resolvedLockedOrientation)) {
+        if (m_page->fullScreenManager() && m_page->fullScreenManager()->isFullScreen()) {
+            if (!m_page->fullScreenManager()->lockFullscreenOrientation(resolvedLockedOrientation)) {
```
Seems we need a `protectedPage()` to call `fullScreenManager()` on.

rniwa replied:
right, fixed.

cdumez commented on Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
```diff
@@ -122,15 +122,15 @@ void WebScreenOrientationManagerProxy::lock(WebCore::ScreenOrientationLockType l
         }
 #endif
-        if (!didLockOrientation && !m_page.uiClient().lockScreenOrientation(m_page, resolvedLockedOrientation)) {
+        if (!didLockOrientation && !m_page->uiClient().lockScreenOrientation(m_page.get(), resolvedLockedOrientation)) {
```
Seems we need to pass a `protectedPage()` to `lockScreenOrientation()`.

rniwa replied:
fixed.

cdumez commented on Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
```diff
@@ -150,13 +150,13 @@ void WebScreenOrientationManagerProxy::unlock()
 #if ENABLE(FULLSCREEN_API)
-    if (m_page.fullScreenManager() && m_page.fullScreenManager()->isFullScreen()) {
-        m_page.fullScreenManager()->unlockFullscreenOrientation();
+    if (m_page->fullScreenManager() && m_page->fullScreenManager()->isFullScreen()) {
```
`fullScreenManager()` is not a trivial function so we need to protected the page.

rniwa replied:
sure, fixed.

cdumez commented on Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
```diff
@@ -150,13 +150,13 @@ void WebScreenOrientationManagerProxy::unlock()
 #endif
     if (!didUnlockOrientation)
-        m_page.uiClient().unlockScreenOrientation(m_page);
+        m_page->uiClient().unlockScreenOrientation(m_page.get());
```
We likely need to pass a `protectedPage()` to `unlockScreenOrientation()`

rniwa replied:
sure, fixed.

cdumez commented on Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp:
```diff
@@ -61,20 +60,22 @@ WebModelPlayerProvider::~WebModelPlayerProvider() = default;
-    if (m_page.corePage()->settings().modelProcessEnabled())
-        return WebProcess::singleton().modelProcessModelPlayerManager().createModelProcessModelPlayer(m_page, client);
+    if (page->corePage()->settings().modelProcessEnabled())
+        return WebProcess::singleton().modelProcessModelPlayerManager().createModelProcessModelPlayer(page.get(), client);
```
We likely don't need `.get()`

cdumez commented on Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp:
```diff
@@ -61,20 +60,22 @@ WebModelPlayerProvider::~WebModelPlayerProvider() = default;
-    if (m_page.useARKitForModel())
-        return ARKitInlinePreviewModelPlayerMac::create(m_page, client);
+    if (page->useARKitForModel())
+        return ARKitInlinePreviewModelPlayerMac::create(page.get(), client);
```
We likely don't need `.get()`

rniwa replied:
ok.

cdumez commented on Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp:
```diff
@@ -61,20 +60,22 @@ WebModelPlayerProvider::~WebModelPlayerProvider() = default;
 #endif
 #if ENABLE(ARKIT_INLINE_PREVIEW_IOS)
-    return ARKitInlinePreviewModelPlayerIOS::create(m_page, client);
+    return ARKitInlinePreviewModelPlayerIOS::create(page.get(), client);
```
We likely don't need `.get()`

rniwa replied:
ok.

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.h:
```diff
@@ -48,26 +48,25 @@ class WebPage;
-    {
-    }
-    
+    WebSpeechSynthesisClient(WebPage&);
```
explicit?

rniwa replied:
sure, fixed.

rniwa committed:
Fix NoUncountedMemberChecker warnings in WebScreenOrientationManagerProxy, WebModelPlayerProvider, and WebSpeechSynthesisClient.
https://bugs.webkit.org/show_bug.cgi?id=279286

Reviewed by Chris Dumez.

Use more smart pointers for member variables in these classes.

* Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.cpp:
(WebKit::WebScreenOrientationManagerProxy::WebScreenOrientationManagerProxy):
(WebKit::WebScreenOrientationManagerProxy::~WebScreenOrientationManagerProxy):
(WebKit::WebScreenOrientationManagerProxy::setCurrentOrientation):
(WebKit::WebScreenOrientationManagerProxy::lock):
(WebKit::WebScreenOrientationManagerProxy::unlock):
* Source/WebKit/UIProcess/WebScreenOrientationManagerProxy.h:
* Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp:
(WebKit::WebModelPlayerProvider::createModelPlayer):
* Source/WebKit/WebProcess/Model/WebModelPlayerProvider.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.cpp:
(WebKit::WebSpeechSynthesisClient::WebSpeechSynthesisClient):
(WebKit::WebSpeechSynthesisClient::protectedPage const):
(WebKit::WebSpeechSynthesisClient::voiceList):
(WebKit::WebSpeechSynthesisClient::corePageObserver const):
(WebKit::WebSpeechSynthesisClient::resetState):
(WebKit::WebSpeechSynthesisClient::speak):
(WebKit::WebSpeechSynthesisClient::cancel):
(WebKit::WebSpeechSynthesisClient::pause):
(WebKit::WebSpeechSynthesisClient::resume):
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.h:
(WebKit::WebSpeechSynthesisClient::WebSpeechSynthesisClient): Deleted.

Canonical link: https://commits.webkit.org/283330@main
